src/components/NewCommentForm.tsx

```diff
@@ -2,7 +1,105 @@
+import {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  FormEvent,
+  SetStateAction,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { addComment } from '../api/comments';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post;
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: FC<Props> = ({ selectedPost, setComments }) => {
+  const [authorName, setAuthorName] = useState('');
+  const [authorNameError, setAuthorNameError] = useState(false);
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [authorEmailError, setAuthorEmailError] = useState(false);
+  const [comentText, setComentText] = useState('');
+  const [commentTextError, setCommentTextError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleInputChange = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    switch (event.target.name) {
+      case 'name':
+        setAuthorNameError(false);
+        setAuthorName(event.target.value.trim());
+        break;
+
+      case 'email':
+        setAuthorEmailError(false);
+        setAuthorEmail(event.target.value.trim());
+        break;
+
+      case 'body':
+        setCommentTextError(false);
+        setComentText(event.target.value.trim());
+        break;
+    }
+  };
+
+  const handleResetForm = () => {
+    setAuthorName('');
+    setAuthorEmailError(false);
+    setAuthorEmail('');
+    setAuthorEmailError(false);
+    setComentText('');
+    setCommentTextError(false);
+  };
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const preparedAuthorName = authorName.trim();
+    const preparedAuthorEmail = authorEmail.trim();
+    const preparedCommentText = comentText.trim();
+
+    if (!preparedAuthorName) {
+      setAuthorNameError(true);
+    }
+
+    if (!preparedAuthorEmail) {
+      setAuthorEmailError(true);
+    }
+
+    if (!preparedCommentText) {
+      setCommentTextError(true);
+    }
+
+    if (!preparedAuthorEmail || !preparedAuthorName || !preparedCommentText) {
+      return;
+    }
+
+    setIsLoading(true);
+
+    const commentToAdd: Omit<Comment, 'id'> = {
+      postId: selectedPost.id,
+      name: preparedAuthorName,
+      email: preparedAuthorEmail,
+      body: preparedCommentText,
+    };
+
+    addComment(commentToAdd)
+      .then(newComment => {
+        setComments(currentComments => [...currentComments, newComment]);
+        setComentText('');
+      })
+      .catch(() => 'Something Wrong')
```

Comment:
The error handling here is not effective. Consider setting an error state to display a message to the user when the comment addition fails.